http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java

```diff
@@ -165,12 +184,51 @@ private HttpURLConnection createDefaultConnection(URI uri, SSLSocketFactory sock
             httpsConnection.setSSLSocketFactory(socketFactory);
         }
 
+        if (proxy.isPresent() && shouldProxyAuthorize()) {
+            connection.addRequestProperty("proxy-authorization", String.format("Basic %s", encodedAuthToken(proxyConfiguration)));
+        }
+
         connection.setConnectTimeout(saturatedCast(options.get(SdkHttpConfigurationOption.CONNECTION_TIMEOUT).toMillis()));
         connection.setReadTimeout(saturatedCast(options.get(SdkHttpConfigurationOption.READ_TIMEOUT).toMillis()));
 
         return connection;
     }
 
+    /**
+     * If a proxy is configured with username+password, then set the proxy-authorization header to authorize ourselves with the
+     * proxy
+     */
+    private static String encodedAuthToken(ProxyConfiguration proxyConfiguration) {
+
+        String authToken = String.format("%s:%s", proxyConfiguration.username(), proxyConfiguration.password());
+        return Base64.getEncoder().encodeToString(authToken.getBytes(StandardCharsets.UTF_8));
+    }
+
+    private boolean shouldProxyAuthorize() {
+        return this.proxyConfiguration != null
+               && ! StringUtils.isEmpty(this.proxyConfiguration.username())
+               && ! StringUtils.isEmpty(this.proxyConfiguration.password());
+    }
+
+    private Optional<Proxy> determineProxy(URI uri) {
+        if (isProxyEnabled() && isProxyHostIncluded(uri)) {
+            return Optional.of(
+                new Proxy(Proxy.Type.HTTP,
+                          new InetSocketAddress(this.proxyConfiguration.host(), this.proxyConfiguration.port())));
+        }
```

Comment:
this resolves the address at construction time, which is not what we want. We should use `InetSocketAddress.createUnresolved()`